qlty-cli/src/commands/fmt.rs

```diff
@@ -73,6 +86,22 @@ impl Fmt {
         }
     }
 
+    fn git_add(&self, paths: &[PathBuf]) -> Result<()> {
+        let mut args = vec!["add"];
+
+        for path in paths {
+            if let Some(path_str) = path.to_str() {
+                args.push(path_str);
+            }
+        }
+
+        if args.len() > 1 {
+            cmd("git", &args).run()?;
```

Comment:
I was unable to find a way to accomplish this same behavior with the git2 crate

Reply:
git2 doesn't support add? What is the missing piece here?

Reply:
It has an add capability on the Index struct, which I thought would provide what we need. Unfortunately in my (admittedly very limited) testing, it did not behave exactly the same as shelling out to git add.

I am fairly confident git2 has the raw functionality but it wasn't immediately obvious to me what sequence of git2 calls emulates the high level command